Mathlib/CategoryTheory/Functor/KanExtension/Adjunction.lean

```diff
@@ -0,0 +1,130 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Functor.KanExtension.Pointwise
+
+/-! # The Kan extension functor
+
+Given a functor `L : C ⥤ D`, we define the left Kan extension functor
+`L.lan : (C ⥤ H) ⥤ (D ⥤ H)` which sends a functor `F : C ⥤ H` to its
+left Kan extension along `L`. This is defined if all `F` have such
+a left Kan extension. It is shown that `L.lan` is the left adjoint to
+the functor `(D ⥤ H) ⥤ (C ⥤ H)` given by the precomposition
+with `L` (see `Functor.lanAdjunction`).
+
+## TODO
+- dualize the results for right Kan extensions
+- refactor the file `CategoryTheory.Limits.KanExtension`
```

Comment:
Can you please add 1 or 2 sentences explaining what this refactor should be doing?

Reply:
Thanks for the suggestion. I have added more details.